apps/emqx_authn/src/emqx_authn_password_hashing.erl

```diff
@@ -108,7 +123,13 @@ hash(#{name := bcrypt, salt_rounds := _} = Algorithm, Password) ->
     Hash = emqx_passwd:hash({bcrypt, Salt0}, Password),
     Salt = Hash,
     {Hash, Salt};
-
+hash(#{name := pbkdf2,
+       mac_fun := MacFun,
+       iterations := Iterations} = Algorithm, Password) ->
+    Salt = gen_salt(Algorithm),
```

Comment:
Just curiosity: in the tests, the salts used are things like `"ATHENA.MIT.EDUraeburn"`, which are not random, but this `gen_salt` clause will generate random bytes.  PBKDF2 does not care if the salt does not have the form of `"ATHENA.MIT.EDUraeburn"`, right? :slightly_smiling_face:

Reply:
Yes, salt is arbitrary. The example was taken from `pbkdf2` own tests.